Services/UserService/UserService.cs

```diff
@@ -0,0 +1,141 @@
+﻿using AutoMapper;
+using webapi.Dtos.Users;
+using webapi.Models;
+
+namespace webapi.Services.UserService
+{
+    /// <summary>
+    /// Implementation of the user service.
+    /// </summary>
+    public class UserService : IUserService
+    {
+        // In-memory storage for users
+        private static List<Users> users = new List<Users>();
+
+        private readonly IMapper _mapper;
+
+        public UserService(IMapper mapper)
+        {
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Retrieves all users.
+        /// </summary>
+        /// <returns>A list of all users.</returns>
+        public async Task<List<UsersDto>> GetAllUsers()
+        {
+            return (users.Select(c => _mapper.Map<UsersDto>(c))).ToList();
+        }
+
+        /// <summary>
+        /// Retrieves a user by ID.
+        /// </summary>
+        /// <param name="id">The ID of the user to retrieve.</param>
+        /// <returns>The user with the specified ID, or null if not found.</returns>
+        public async Task<UsersDto> GetUserById(int id)
+        {
+            try
+            {
+                UsersDto result = _mapper.Map<UsersDto>(users.FirstOrDefault(c => c.Id == id));
```

Comment:
why are you sending password also in get user by id?

Reply:
done